src/EnergyPlus/Furnaces.cc

```diff
@@ -9445,7 +9716,59 @@ namespace Furnaces {
 			} else {
 				FanSpeedRatio = CompOnFlowRatio;
 			}
-		} else {
+		}
+		else if (true == Furnace(FurnaceNum).bIsIHP)
+		{
+			if (!CurDeadBandOrSetback(Furnace(FurnaceNum).ControlZoneNum) && present(SpeedNum)){
+			//if(present(SpeedNum)) {
+				CompOnMassFlow = GetAirMassFlowRateIHP(Furnace(FurnaceNum).CoolingCoilIndex, SpeedNum, SpeedRatio, false);
+				CompOnFlowRatio = CompOnMassFlow /
+					GetAirMassFlowRateIHP(Furnace(FurnaceNum).CoolingCoilIndex, GetMaxSpeedNumIHP(Furnace(FurnaceNum).CoolingCoilIndex), 1.0, false);
+				MSHPMassFlowRateLow = GetAirMassFlowRateIHP(Furnace(FurnaceNum).CoolingCoilIndex, SpeedNum, 0.0, false);
+				MSHPMassFlowRateHigh = GetAirMassFlowRateIHP(Furnace(FurnaceNum).CoolingCoilIndex, SpeedNum, 1.0, false);
+			}
+
+			// Set up fan flow rate during compressor off time
+			if (Furnace(FurnaceNum).OpMode == ContFanCycCoil && present(SpeedNum)) {
+				if (Furnace(FurnaceNum).AirFlowControl == UseCompressorOnFlow && CompOnMassFlow > 0.0) {
+					CompOffMassFlow = GetAirMassFlowRateIHP(Furnace(FurnaceNum).CoolingCoilIndex, SpeedNum, 1.0, false);
+					CompOffFlowRatio = CompOffMassFlow /
+						GetAirMassFlowRateIHP(Furnace(FurnaceNum).CoolingCoilIndex, GetMaxSpeedNumIHP(Furnace(FurnaceNum).CoolingCoilIndex), 1.0, false);
+				}
+			}
```

Comment:
Lot's of GetAirMassFlowRateIHP calls here. Should have an array of Mdot vs speed num to avoid excess function calls. Did you look at SetOnOffMassFlowRate or SetOnOffMassFlowRateVSCoil to see that MSHPassFlowRateLow/High are set there automatically?